addons/project_timesheet_holidays/models/hr_employee.py

```diff
@@ -21,20 +21,22 @@ def create(self, vals_list):
         return employees
 
     def write(self, vals):
+        if vals.get('active'):
+            inactive_emp = self.filtered(lambda e: not e.active)
         result = super(Employee, self).write(vals)
         self_company = self.with_context(allowed_company_ids=self.company_id.ids)
         if 'active' in vals:
             if vals.get('active'):
                 # Create future holiday timesheets
-                inactive_emp = self_company.filtered(lambda e: not e.active)
-                inactive_emp._create_future_public_holidays_timesheets(self)
+                inactive_emp = inactive_emp.with_env(self_company.env)
+                inactive_emp._create_future_public_holidays_timesheets(inactive_emp)
```

Comment:
Was also mislead in https://github.com/odoo/odoo/commit/9a22fec34c293b001c3bae557cb2965ade321b5b by the fact that method expect employee recordset to be passed as parameter :man_facepalming:.

I do not want to change the `_create_future_public_holidays_timesheets()` method signature for stable versions, maybe we could update it to use `self` instead of the `employees` parameter on master ?

Reply:
Yep indeed, I agree. 😅 

Thanks!